### DISCUSSION TITLE
Sorting the results of orderless matching

### DISCUSSION THREAD

**Discussion by ghost** [minad/vertico#541] [Q&A]
The vast majority of my completion attempts involve targeting a symbol / identifier / ... that I already know, and whose abbreviated components I enter in left-to-right order. In such cases, orderless matching can return worthless "noise" matches.

What I would like is to be able to post-filter orderless's matches, keeping
- Only results where the components were matched in strict left-to-right order and where the first component match was prefix, anchored on the left
- Only results where the components were matched in strict left-to-right order
- All results, irrespective of the order in which components were matched

Is such post-filtering possible?

**Comment by ghost** [comment]
This issue (now discussion) reflected my misunderstanding of emacs's completion architecture.  I have been exploring additional resources and am beginning to see the larger picture, especially the role of earlier completion styles.

**Reply by minad** [reply]
Orderless itself intentionally doesn't provide any sorting, and it is good like this. But you can write a custom completion style which sorts or you could hack something with Orderless and `vertico-sort-function` or `vertico-sort-override-function`.
